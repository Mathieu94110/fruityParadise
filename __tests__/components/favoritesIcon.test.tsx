import { render, screen, fireEvent } from '@testing-library/react-native';
import FavoritesIcon from '@/components/favoritesIcon';

describe('favoritesIcon', () => {
  const mockFn = jest.fn();

  const setup = () => {
    const utils = render(
      <FavoritesIcon isFavorite={false} handleFavorite={mockFn} />,
    );
    return { ...utils };
  };

  it('should favoritesIcon render correctly', () => {
    const tree = setup();
    expect(tree).toMatchSnapshot();
  });

  it('should having by default #FF385C color', () => {
    render(<FavoritesIcon isFavorite={true} handleFavorite={mockFn} />);
    const starIcon = screen.getByTestId('star-icon');
    expect(starIcon).toHaveStyle({ color: '#FF385C' });
  });

  it('should on icon click handleFavorite have been called and icon now having #1A1A1A color', () => {
    const { getByTestId } = setup();
    const starIcon = getByTestId('star-icon');
    fireEvent.press(starIcon);
    expect(mockFn).toHaveBeenCalled();
    expect(starIcon).toHaveStyle({ color: '#1A1A1A' });
  });
});
