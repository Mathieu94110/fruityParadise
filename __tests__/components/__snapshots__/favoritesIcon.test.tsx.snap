// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`favoritesIcon render correctly 1`] = `
<View>
  <Text />
</View>
`;
